exoscale/api/compute.py

```diff
@@ -0,0 +1,2088 @@
+        self.icmp_code = None
+        self.icmp_type = None
+
+    def _parse_port(self):
+        """
+        Parse the rule port attribute and returns a port range.
+
+        Returns:
+            int: start port
+            int: end port
+        """
+
+        if not self.port:
+            return None, None
+
+        start_port, end_port = (
+            self.port.split("-", maxsplit=1) if "-" in self.port else (self.port, None)
+        )
+        if not end_port:
+            end_port = start_port
+
+        return int(start_port), int(end_port)
+
+
+@attr.s
+class SSHKey(Resource):
+    """
+    A SSH key.
+
+    Attributes:
+        name (str): the SSH key unique name
+        fingerprint (str): the SSH key fingerprint
+        private_key (str): the SSH private key, or None if registered SSH key
+    """
+
+    compute = attr.ib(repr=False)
+    res = attr.ib(repr=False)
+    name = attr.ib()
+    fingerprint = attr.ib()
+    private_key = attr.ib(default=None, repr=False)
+
+    @classmethod
+    def from_cs(cls, compute, res):
+        return cls(
+            compute,
+            res,
+            name=res["name"],
+            fingerprint=res["fingerprint"],
+            private_key=res.get("privatekey", None),
+        )
+
+    def delete(self):
+        """
+        Delete the SSH key.
+
+        Returns:
+            None
+        """
+
+        try:
+            self.compute.cs.deleteSSHKeyPair(name=self.name)
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"], e.error)
+
+        self.res = None
+        self.name = None
+        self.fingerprint = None
+        self.private_key = None
+
+
+@attr.s
+class Zone(Resource):
+    """
+    An Exoscale zone.
+
+    Attributes:
+        id (str): the zone unique identifier
+        name (str): the zone name
+    """
+
+    res = attr.ib(repr=False)
+    id = attr.ib()
+    name = attr.ib()
+
+    @classmethod
+    def from_cs(cls, res):
+        return cls(res, id=res["id"], name=res["name"])
+
+
+class ComputeAPI(API):
+    """
+    An Exoscale Compute API client.
+
+    Parameters:
+        key (str): the Compute API key
+        secret (str): the Compute API secret
+        endpoint (str): the Compute API endpoint
+        trace (bool): API request/response tracing flag
+    """
+
+    def __init__(self, key, secret, endpoint=None, trace=False):
+        endpoint = "https://api.exoscale.com/compute" if endpoint is None else endpoint
+        super().__init__(endpoint, key, secret, trace)
+
+        self.cs = CloudStack(
+            key=key,
+            secret=secret,
+            endpoint=endpoint,
+            headers={"User-Agent": API.user_agent},
+            fetch_result=True,
+            trace=self.trace,
+        )
+
+    def __repr__(self):
+        return "ComputeAPI(endpoint='{}' key='{}')".format(self.endpoint, self.key)
+
+    def __str__(self):
+        return self.__repr__()
+
+    ### Anti-Affinity Group
```

Comment:
I get PEP8 warnings for comments that start with more than a `#`.

I don't have any feelings eitherway about this but we might wish to line up with the established conventions on that one.

Reply:
Well I went with [Black](https://black.readthedocs.io/en/stable/) on this project since it's also used in some of our other internal projects, which uses a different set of rules than *pycodestyle* ([the new](https://pycodestyle.readthedocs.io/en/latest/) `pep8`), so as long as Black doesn't yell/reformats I'm good with it :)